running...
Downloading: "https://download.pytorch.org/models/resnet18-5c106cde.pth" to /home/user/.cache/torch/checkpoints/resnet18-5c106cde.pth
  0%|          | 0.00/44.7M [00:00<?, ?B/s]  5%|▍         | 2.08M/44.7M [00:00<00:02, 21.7MB/s] 13%|█▎        | 5.88M/44.7M [00:00<00:01, 32.3MB/s] 27%|██▋       | 12.0M/44.7M [00:00<00:00, 46.7MB/s] 49%|████▉     | 21.8M/44.7M [00:00<00:00, 68.6MB/s] 74%|███████▍  | 33.0M/44.7M [00:00<00:00, 86.1MB/s] 99%|█████████▊| 44.0M/44.7M [00:00<00:00, 93.4MB/s]100%|██████████| 44.7M/44.7M [00:00<00:00, 76.2MB/s]Files already downloaded and verified
Files already downloaded and verified
Files already downloaded and verified
not sorted
anderes: 
not sorted
anderes: 
IMAGENET 

im using:cuda:0
Epoch 0/2
----------

Loss: 3.3025 Acc: 0.0000 Uncertainty_mean: 0.7857 
Saved: ./results/models/ResNet18IMAGENET_TRAINcrossEntropyPretrained/00.pth

Epoch 1/2
----------
Loss: 2.2408 Acc: 0.0000 Uncertainty_mean: 0.8671 
Saved: ./results/models/ResNet18IMAGENET_TRAINcrossEntropyPretrained/01.pth

Epoch 2/2
----------
Loss: 0.7770 Acc: 1.0000 Uncertainty_mean: 0.8882 
Saved: ./results/models/ResNet18IMAGENET_TRAINcrossEntropyPretrained/02.pth

Training complete in 0m 4s
Best Acc: 1.000000 Best Uncertainty: 0.785748
Loading model ./results/models/ResNet18IMAGENET_TRAINcrossEntropyPretrained/00.pth
Traceback (most recent call last):
  File "codebase/main.py", line 275, in <module>
    main()
  File "codebase/main.py", line 272, in main
    runExperiments()
  File "codebase/main.py", line 268, in runExperiments
    train_ImagenetLvl1(train= True, criterion_name = "crossEntropy")
  File "codebase/main.py", line 125, in train_ImagenetLvl1
    defineExperiment(modelList, criterion_name=criterion_name, optimizer=optimizer, train_dataloader=dataloaders["IMAGENET_TRAIN_LVL1"], num_train_classes =19 , test_dataloader=dataloaders["IMAGENET_TEST_LVL1"], num_test_classes=19 ,train=train, pretrained =True, num_epochs = 3, uncertaintyThreshold = -0.1)
  File "codebase/main.py", line 89, in defineExperiment
    val_acc_hist, uncertainty_histry = eval_model(modelList, test_dataloader, model_directory ,device, num_classes = num_test_classes, hierarchicalModelPathList =hierarchicalModelPathList , train_dataloader= train_dataloader , test_dataloader =test_dataloader)
  File "/home/user/experiments/codebase/eval.py", line 176, in eval_model
    inputs, labels, outputs, preds, running_corrects, u, u_mean, epoch_acc, epoch_uncertainty,  correctWhileStaySuper, correctWhileLeaveSuper, falseWhileStaySuper ,falseWhileLeaveSuper = calculate_results()
  File "/home/user/experiments/codebase/eval.py", line 148, in calculate_results
    if calculate_confusion_Matrix:
UnboundLocalError: local variable 'calculate_confusion_Matrix' referenced before assignment
